### ISSUE TITLE
Automate the update of truststore in travisci

### ISSUE THREAD

**Issue by jvehent** [mozilla/tls-observatory#133]
Updating the truststores in the docker container could be done automatically during build time, so we always have the latest version when we build the containers.

**Comment by 0xdiba** [comment]
Should an updated truststore trigger a rescan of all domains?

**Comment by jvehent** [comment]
No. This is too costly and brings no benefit without notifying the users who triggered the original scan. I think it's better handled by daily rescans from people who care to do so.

**Comment by jvehent** [comment]
Rather than using travisci, we could have a job that reconcile the database with data in NSS and the CA program. See https://bugzilla.mozilla.org/show_bug.cgi?id=1295698

**Comment by tomato42** [comment]
What about stuff like WoSign?

**Comment by jvehent** [comment]
> What about stuff like WoSign?

What exactly are you referring to? WoSign is trusted at the moment. We should update the truststores when it is no longer trusted.

**Comment by tomato42** [comment]
I mean that we can have situation in which certificates which were signed by a particular CA only before certain date are trusted. There are other instances of this, but I don't remember their names now.

**Comment by jvehent** [comment]
I think it's probably its own separate issue. Maybe an analyzer that calls moz::pkix? This issue is specifically about keeping the list of CAs up to date in the database.

**Comment by tomato42** [comment]
Yes, but I wonder if such CAs shouldn't be tagged, so that he UI could show that the server certificate may not be trusted.

In other words, to not have a situation in which Firefox says the certificate is invalid but the scanner says everything is hunky dory, without even a hint that something may be wrong.

**Comment by jvehent** [comment]
@tomato42 : let's take this to #151.

**Comment by jvehent** [comment]
I have an intermediate solution in https://github.com/mozilla/tls-observatory/pull/157/commits/40ad9cf1d9a7a2a5a82f85119ec63cbe32c230f0
Travisci pulls the latest from https://github.com/kirei/catt and updates the truststores in configuration, then scanners disable the trust of certificates no longer in truststores.

This still depends on building new scanners and restarting them, which I dislike. This issue will really be fixed when we can have a daily job that updates the DB based on data from the root CA program.

**Comment by jvehent** [comment]
`catt` continues to be very out of date, so today I added this tool: https://github.com/mozilla/tls-observatory/blob/master/tools/retrieveMozillaTrustedCAs.go
It retrieves the PEM of trusted mozilla CAs directly from the common CA database, which is the most up to date source of information. We just need to make sure there are no differences between this and the content of NSS.

**Comment by jvehent** [comment]
2 out of 4!
Microsoft truststore is now also being pulled from the Common CA database via https://github.com/mozilla/tls-observatory/blob/master/tools/retrieveTruststoreFromCADatabase.go
